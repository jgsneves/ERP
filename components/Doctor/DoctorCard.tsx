import { Grid, Avatar } from "@chakra-ui/react";
import Link from "next/link";
import { Text } from "@chakra-ui/react";

interface Props {
  name: string;
  id: string;
  CRM: string | null;
  medicalCompany: string;
}

export default function DoctorCard({ id, name, CRM, medicalCompany }: Props) {
  return (
    <Link href={`/medicos/${id}`}>
      <Grid
        borderRadius="lg"
        padding={2}
        cursor="pointer"
        _hover={{
          backgroundColor: "gray.200",
        }}
      >
        <Avatar name={name} src={undefined} />
        <Text>Nome: {name}</Text>
        <Text>CRM: {CRM}</Text>
        <Text>Empresa m√©dica: {medicalCompany}</Text>
      </Grid>
    </Link>
  );
}
