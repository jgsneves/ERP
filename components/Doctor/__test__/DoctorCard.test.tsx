import { render } from "@testing-library/react";
import DoctorCard from "../DoctorCard";

describe("DoctorCard test suite", () => {
  it("should render", () => {
    const { asFragment } = render(
      <DoctorCard
        id="1"
        CRM="2020"
        name="José da Silva"
        medicalCompany="ABCD Empresa Médica"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
