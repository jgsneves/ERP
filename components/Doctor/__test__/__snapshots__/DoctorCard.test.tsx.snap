// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoctorCard test suite should render 1`] = `
<DocumentFragment>
  <a
    href="/medicos/1"
  >
    <div
      class="css-1naneae"
    >
      <span
        class="chakra-avatar css-1ruhx4w"
      >
        <div
          aria-label="José da Silva"
          class="chakra-avatar__initials css-0"
          role="img"
        >
          Jd
        </div>
      </span>
      <p
        class="chakra-text css-0"
      >
        Nome: José da Silva
      </p>
      <p
        class="chakra-text css-0"
      >
        CRM: 2020
      </p>
      <p
        class="chakra-text css-0"
      >
        Empresa médica: ABCD Empresa Médica
      </p>
    </div>
  </a>
</DocumentFragment>
`;
